external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -330,27 +341,27 @@ private[sql] class AvroDeserializer(
     var i = 0
     while (i < length) {
       val sqlField = sqlType.fields(i)
-      val avroField = avroType.getField(sqlField.name)
```

Comment:
is the original code `avroType.getField(sqlField.name)` also a O(n) linear lookup?

Reply:
No, it is a constant-time map lookup (from `org.apache.avro.Schema.RecordSchema`):
```
    public Field getField(String fieldname) {
      if (fieldMap == null)
        throw new AvroRuntimeException("Schema fields not set yet");
      return fieldMap.get(fieldname);
    }
```
I think this is not so performance-critical given that it is executed only once per query? It is not done on a per-record basis.